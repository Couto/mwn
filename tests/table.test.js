const { expect } = require('./test_wiki');
const { ElementWithAttributes, Table, TableRow, TableCell, TableHeader, TableCaption } = require('../build/wikitext/table');

describe('table', function () {
	it('element classes', () => {
		let element = new ElementWithAttributes();
		element.setAttributes({
			scope: 'col',
			class: 'class1'
		});
		element.setClasses(['class2', 'class3']);
		expect(element.attributesToText()).to.equal('scope="col" class="class2 class3 class1"');
	});

	it('element styles', () => {
		let element = new ElementWithAttributes();
		element.setAttributes({
			scope: 'col',
			style: 'width: 4em'
		});
		element.setClasses(['class1', 'class2']);
		element.setStyle('padding', '0');
		expect(element.attributesToText()).to.equal('scope="col" style="width: 4em; padding: 0;" class="class1 class2"');
	});

	it('table', () => {
		var table = new Table();
		table.setHeaders(['Header text', 'Header text', 'Header text']);
		table.addRow(['Example', 'Example', 'Example']);
		table.addRow(['Example', 'Example', 'Example']);
		expect(table.toText()).toMatchSnapshot();

		table.transformColumn(1, text => text + ' ' + text);
		expect(table.toText()).toMatchSnapshot();

		table.addColumn('New column', ['c1', 'c2'], 2);
		expect(table.toText()).toMatchSnapshot();
	});

	it('table from text', () => {

	});
});
