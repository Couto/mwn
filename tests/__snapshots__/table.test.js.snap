// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table table 1`] = `
"{| class=\\"wikitable sortable\\"
|-
! scope=\\"col\\" | Header text
! scope=\\"col\\" | Header text
! scope=\\"col\\" | Header text
|- 
| Example
| Example
| Example
|- 
| Example
| Example
| Example
|}"
`;

exports[`table table 2`] = `
"{| class=\\"wikitable sortable\\"
|-
! scope=\\"col\\" | Header text
! scope=\\"col\\" | Header text
! scope=\\"col\\" | Header text
|- 
| Example
| Example Example
| Example
|- 
| Example
| Example Example
| Example
|}"
`;

exports[`table table 3`] = `
"{| class=\\"wikitable sortable\\"
|-
! scope=\\"col\\" | Header text
! scope=\\"col\\" | Header text
! scope=\\"col\\" | Header text
! scope=\\"col\\" | New column
|- 
| Example
| Example Example
| c1
| Example
|- 
| Example
| Example Example
| c2
| Example
|}"
`;
